### PR TITLE
FAT-128 FAT-120 FAT-121 Prompt language change after live language change

### PR DESCRIPTION
**Pull request by grsoares21**
### 📝 Description
This PR adds a few prompts suggesting to the user to change their device language. The prompts appear when:
- The user changes Live language and the target language is available for the last seen device
- The user updates the devices to a firmware that supports the current Live language
- The user had a non-English language installed on their device before a a firmware update (the language gets automatically reinstalled since it's deleted by the update)

### ❓ Context
[FAT-128]
[FAT-121]
[FAT-120]

### ✅ Checklist
- [x] **Test coverage** 
- [x] **Atomic delivery** 
- [x] **No breaking changes** 

### 📸 Demo
Prompt after Live language change:
prompt-after-live-language-change
Prompt when Live's language first becomes available for the device:
prompt-after-live-update
Reinstall language after firmware update:
reinstall-after-live-update

### 🚀 Expectations to reach

[FAT-128]: https://ledgerhq.atlassian.net/browse/FAT-128?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FAT-121]: https://ledgerhq.atlassian.net/browse/FAT-121?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[FAT-120]: https://ledgerhq.atlassian.net/browse/FAT-120?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ledger-live-desktop/tests/specs/manager/devicelocalization.spec.ts [hunk 1]
```diff
@@ -1,14 +1,17 @@
 import test from "../../fixtures/common";
 import { expect } from "@playwright/test";
 import { ManagerPage } from "../../models/ManagerPage";
+import { LanguageInstallation } from "../../models/LanguageInstallation";
 import { DeviceAction } from "../../models/DeviceAction";
 import { Layout } from "../../models/Layout";
+import { languageIds } from "@ledgerhq/types-live";
```
Comment (alexandremgo) [inline apps/ledger-live-desktop/tests/specs/manager/devicelocalization.spec.ts:7]: `languageIds` not used

File: apps/ledger-live-desktop/tests/models/LanguageInstallation.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Page, Locator } from "@playwright/test";
+
+export class LanguageInstallation {
+  readonly page: Page;
+
+  readonly installLanguageButton: Locator;
+  readonly closeLanguageInstallationButton: Locator;
+  readonly languageInstalled: Locator;
```
[the remark this answers is not in the document]
Comment (grsoares21) [reply apps/ledger-live-desktop/tests/models/LanguageInstallation.ts]: Fixed

File: apps/ledger-live-desktop/tests/specs/manager/devicelocalization.spec.ts (hunk 1 above)
Comment (grsoares21) [reply apps/ledger-live-desktop/tests/specs/manager/devicelocalization.spec.ts]: Fixed

File: apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/steps/02-step-device-language.tsx [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    installLanguage,
+    deviceLocalizationFeatureFlag?.enabled,
+    transitionTo,
+  ]);
+
+  const deviceName = getDeviceModel(device.modelId).productName;
+
+  return (
+    <Flex justifyContent="center" flex={1}>
+      {isLanguagePromptOpen && !installingLanguage && (
```
Comment (alexandremgo) [inline apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/steps/02-step-device-language.tsx:97]: just to be safe in the future, i think it's better to user ternary operators than logical ones inside JSX

File: apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx [hunk 4]
```diff
@@ -132,10 +150,19 @@ const UpdateModal = ({
       } else {
         steps.push(updatingStep);
       }
+
+      if (
+        firmware &&
+        isDeviceLocalizationSupported(firmware.final.name, deviceModelId) &&
+        deviceLocalizationFeatureFlag.enabled
```
Comment (alexandremgo) [inline apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx:157]: `deviceLocalizationFeatureFlag?.enabled`;)

File: apps/ledger-live-desktop/src/renderer/screens/manager/DeviceStorage/DeviceLanguageInstallation.tsx [hunk 5]
```diff
@@ -1,114 +1,100 @@
-import React, { useState, useCallback, useEffect } from "react";
-import { Button, Flex, Icons, Drawer, Radio, BoxedIcon, Divider, Log } from "@ledgerhq/react-ui";
+import React, { useState, useCallback, useMemo } from "react";
+import { Button, Flex, Icons, Drawer, Radio, Divider, Text } from "@ledgerhq/react-ui";
 import { DeviceInfo, Language } from "@ledgerhq/types-live";
+import { track } from "~/renderer/analytics/segment";
```
Comment (alexandremgo) [inline apps/ledger-live-desktop/src/renderer/screens/manager/DeviceStorage/DeviceLanguageInstallation.tsx:4]: relative import should go after extern import, don't know why my linter is not complaining though 🤷

File: apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/steps/02-step-device-language.tsx (hunk 3 above)
Comment (grsoares21) [reply apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/steps/02-step-device-language.tsx]: Fixed

File: apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx (hunk 4 above)
Comment (grsoares21) [reply apps/ledger-live-desktop/src/renderer/modals/UpdateFirmwareModal/index.tsx]: Fixed

File: apps/ledger-live-desktop/src/renderer/screens/manager/DeviceStorage/DeviceLanguageInstallation.tsx (hunk 5 above)
Comment (grsoares21) [reply apps/ledger-live-desktop/src/renderer/screens/manager/DeviceStorage/DeviceLanguageInstallation.tsx:4]: I don't think `~/...` is considered to be a relative import

Comment (alexandremgo) [reply apps/ledger-live-desktop/src/renderer/screens/manager/DeviceStorage/DeviceLanguageInstallation.tsx:4]: ah that's probably why ! ok

File: apps/ledger-live-desktop/src/renderer/screens/settings/sections/General/ChangeDeviceLanguagePromptDrawer.tsx [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+import { DeviceInfo } from "@ledgerhq/types-live";
+import { setLastSeenDevice } from "~/renderer/actions/settings";
+import { useTranslation } from "react-i18next";
+import { track } from "~/renderer/analytics/segment";
+import { withV3StyleProvider } from "~/renderer/styles/StyleProviderV3";
+import { command } from "~/renderer/commands";
+import { getCurrentDevice } from "~/renderer/reducers/devices";
+import ChangeDeviceLanguagePrompt from "~/renderer/components/ChangeDeviceLanguagePrompt";
+import { DeviceModelId, getDeviceModel } from "@ledgerhq/devices";
+
```
Comment (alexandremgo) [inline apps/ledger-live-desktop/src/renderer/screens/settings/sections/General/ChangeDeviceLanguagePromptDrawer.tsx:16]: We should import `DeviceModelId` enum from `@ledgerhq/types-devices` now ;)

**Review by alexandremgo — APPROVED** [review]
🚀

File: apps/ledger-live-desktop/src/renderer/screens/settings/sections/General/ChangeDeviceLanguagePromptDrawer.tsx (hunk 6 above)
Comment (grsoares21) [reply apps/ledger-live-desktop/src/renderer/screens/settings/sections/General/ChangeDeviceLanguagePromptDrawer.tsx]: fixed

**Pull request merged**